aeron-driver/src/main/java/io/aeron/driver/DriverConductor.java

```diff
@@ -1764,15 +1764,44 @@ private NetworkPublication newNetworkPublication(
         }
     }
 
+    private int getSndbufLength(final UdpChannelTransport channelEndpoint)
+    {
+        final int sndbufLength = channelEndpoint.socketSndbufLength();
+        if (sndbufLength == 0)
+        {
+            return ctx.osDefaultSocketSndbufLength();
+        }
+        else
+        {
+            return sndbufLength;
+        }
+    }
```

Comment:
There is already a method `io.aeron.driver.media.UdpChannel#socketSndbufLengthOrDefault`. However, getting the correct value is more nuanced, i.e.:
* For network publication it should be taken from `SendChannelEndpoint#socketSndbufLength` instead.
* For IPC publication the socket buffers make no sense.
* For publication image the values should be taken from the `ReceiveChannelEndpoint`.

Reply:
I just figured out that Aeron will detect setting a socket buffer length that is larger than the maximum.

```
WARNING: Could not set desired SO_SNDBUF, adjust OS to allow aeron.socket.so_sndbuf attempted=67108864, actual=212992
WARNING: Could not set desired SO_RCVBUF, adjust OS to allow aeron.socket.so_rcvbuf attempted=67108864, actual=212992
```

Reply:
The problem with the channelEndpoint.socketRcvbufLength() and channelEndpoint.socketSndbufLength() is that these methods can't be relied upon in the case that the configured value is larger than the OS maximum. This is the value that is configured; but not the value that is actually used.

The code to get the actual values:
```
           System.out.println("actual.sndBufLength:" + sendDatagramChannel.socket().getSendBufferSize());
            System.out.println("actual.rcvBufLength:" + receiveDatagramChannel.socket().getReceiveBufferSize());
```

And the corresponding output:

```
channelEndpoint.socketRcvbufLength:67108864
channelEndpoint.socketSndbufLength:67108864
actual.sndBufLength:212992
actual.rcvBufLength:212992
```

So the actual values are the value that need to be published in the log buffer. WDYT?

Reply:
@pveentjer [UdpChannelTransport#openDatagramChannel](https://github.com/real-logic/aeron/blob/2f8e497a97b850e2786f6a65397ead859ccdb680/aeron-driver/src/main/java/io/aeron/driver/media/UdpChannelTransport.java#L236-L244) is where we assign `SO_SNDBUF` and `SO_RCVBUF`.
I don't know if it is possible to get the actual values used after the configuration completes (at least from the Java API).

Reply:
I think we should go with the current approach.

Reply:
To get the actual values, use:

```
    System.out.println("actual.sndBufLength:" + sendDatagramChannel.socket().getSendBufferSize());
    System.out.println("actual.rcvBufLength:" + receiveDatagramChannel.socket().getReceiveBufferSize());
```